packages/arc-auth/cdk/src/env.ts

```diff
@@ -0,0 +1,81 @@
+import {DataAwsSecurityGroup} from '@cdktf/provider-aws/lib/data-aws-security-group';
+import {DataAwsSsmParameter} from '@cdktf/provider-aws/lib/data-aws-ssm-parameter';
+import {DataAwsSubnets} from '@cdktf/provider-aws/lib/data-aws-subnets';
+import {TerraformStack} from 'cdktf';
+import {readFileSync} from 'fs';
+
+export const env = {
+  AWS_REGION: "",
+  DB_HOST: "",
+  DB_PORT: 5432,
+  DB_USER: "",
+  DB_PASSWORD: "",
+  DB_DATABASE: "",
+  DB_SCHEMA: "",
+  JWT_SECRET: "",
+  ACM_CERTIFICATE_ARN: "",
+  HOSTED_ZONE_ID: "",
+  DOMAIN_NAME: "",
+  NAMESPACE: "",
+  ENV: "",
+  S3_BUCKET: ""
+};
+
+interface EnvVar {
+  [key: string]: string;
+}
+
+export const getSubnetIds = (scope: TerraformStack) => {
+  const subnets = new DataAwsSubnets(scope, "private_subnets", {
+    filter: [
+      {
+        name: "tag:Name",
+        values: ['demoTagName'],    //Replace demoTagName by Name Tag of subnet id
+      },
+    ],
+  });
+  return subnets.ids;
+}
+
+export const getSecurityGroup = (scope: TerraformStack) => {
+  const sgroup = new DataAwsSecurityGroup(scope, "security_group", {
+    filter: [
+      {
+        name: "tag:Name",
+        values: ['demoTagName'],    //Replace demoTagName by Name Tag of security group
+      },
+    ],
+  });
+  return [sgroup.id];
+};
+
+
+export const getEnv = (scope: TerraformStack) => {
+  let envVar: EnvVar = {};
+  checkEnv();
+
+  for (const key in process.env) {
+    // Check if the property is directly defined on the object (not inherited)
+    if (process.env.hasOwnProperty(key)) {
+      //read value from ssm
+      const ssm = new DataAwsSsmParameter(scope, "db_admin_username_ssm_param", {
+        name: process.env[key] ?? '',
+        withDecryption: true
+      });
+      // Copy the value from process.env to envVar
+      envVar[key] = ssm.value;
+    }
+  }
+
+  return envVar;
+}
+
+
+export const checkEnv = () => {
+  let envToCheck = readFileSync('../.env.schema', "utf8").split(/[\n =]/).filter(Boolean);
```

Comment:
This misses the case when there are comments in the env.
There can be both inline and a whole line comments.
I think the regex is already splitting the string on space, new line and = so you just need to filter values that begin with #.
readFileSync(path, "utf8").split(/[\n =]/).filter(Boolean).filter((elem) => !elem.startsWith("#"));

also I think it would be better to split it into lines by splitting it first into "\n" and then take the left side of equal (i.e. line[0]), because this code assumes that the values will not be there and the values will be filtered using filter(Boolean) so in case the user has a value with a property in the .env.schema this code will treat that as property also. You don't need to deal with values anyway, you just need properties.

you can use this 
const lines = readFileSync(path, "utf-8").split("\n").filter((line) => line.length);
const properties = lines.map((line) => line.split("=")[0]).filter((line) => !line.startsWith("#"));

Reply:
hmm ,
better

Reply:
Why would you always assume a .env file? We should just be checking for environment variables. Assuming an ENV assumes a CI/CD process creates a file when it does

Reply:
Ok then its even easy. We can just use Object.keys(process.env)
just need to take care of extra properties that come in process.env

Reply:
@jamescrowley321 so we can easily assume that the properties that user wants to fetch from SSM. User will have to specify those beforehand.
So we can create an array propertiesToFetch and user have to add properties there
whatever properties user will add we'll fetch it from ssm.

Reply:
We can use `dotenv`. Let's not hand write anything to read an env file or environment variables.